src/translib/transformer/xlate_to_db.go

```diff
@@ -90,6 +90,33 @@ func xfmrHandler(inParams XfmrParams, xfmrFuncNm string) (map[string]map[string]
         return nil, nil
 }
 
+func keyXfmrHandler(inParams XfmrParams, xfmrFuncNm string) (string, error) {
+        log.Info("Received inParams ", inParams, "key transformer function name ", xfmrFuncNm)
+        ret, err := XlateFuncCall(yangToDbXfmrFunc(xfmrFuncNm), inParams)
+	retVal := ""
+        if err != nil {
+                return retVal, err
+        }
+
+        if ((ret != nil) && (len(ret)>0)) {
+                if len(ret) == 2 {
+                        // key xfmr returns err as second value in return data list from <xfmr_func>.Call()
+                        if ret[1].Interface() != nil {
```

Comment:
YangToDB keyXfmr retruns a string and err. Are we asking customer callback to return array of 2 strings where the 2nd one is error string? It looks not intuitive if it is.

Reply:
No we are not asking customer to return array of 2 strings. Transformer infra calls keyXfmr with the keyXfmr name-string combined  GOLANG <kyxfmrNmStr>.**Call()** function. 
**.Call()** returns the return args string and err from keyXfmr packed inside an array.So first arg from this array is the string and second arg is err.